glean-core/src/database/mod.rs

```diff
@@ -707,4 +810,52 @@ mod test {
             );
         }
     }
+
+    #[test]
+    fn test_deferred_ping_lifetime_collection() {
+        // Init the database in a temporary directory.
+        let dir = tempdir().unwrap();
+        let str_dir = dir.path().display().to_string();
+        let db = Database::new(&str_dir, true).unwrap();
+
+        assert!(db.ping_lifetime_data.is_some());
+
+        // Attempt to record a known value.
+        let test_value = "test-value";
+        let test_storage = "test-storage1";
+        let test_metric_id = "telemetry_test.test_name";
+        db.record_per_lifetime(
+            Lifetime::Ping,
+            test_storage,
+            test_metric_id,
+            &Metric::String(test_value.to_string()),
+        )
+        .unwrap();
+
+        // Verify that the data is correctly recorded.
+        let mut found_metrics = 0;
+        let mut snapshotter = |metric_name: &[u8], metric: &Metric| {
+            found_metrics += 1;
+            let metric_id = String::from_utf8_lossy(metric_name).into_owned();
+            assert_eq!(test_metric_id, metric_id);
+            match metric {
+                Metric::String(s) => assert_eq!(test_value, s),
+                _ => panic!("Unexpected data found"),
+            }
+        };
+
+        db.iter_store_from(Lifetime::Ping, test_storage, None, &mut snapshotter);
+        assert_eq!(1, found_metrics, "We only expect 1 Lifetime.Ping metric.");
+
+        let store: SingleStore = unwrap_or!(
```

Comment:
We discussed over Slack and agreed to slightly change this to make sure that something is being written to rkv at all :)

Reply:
Would the comment I added suffice here or do you mean for me to add the whole experiment I did to lowre my confusion?

I personally don't think the experiment adds to test and event makes it a little confusing (since it is code that is not related), but I want to know your opinion :)

Reply:
No, that's fine :)